### PR TITLE
Update iterables

### PR DESCRIPTION
**Pull request by jonathantneal**
- consistently calculate whole iterable lengths
- replace most references to arraylike with iterable
- correct arraylike bug in Promise
- add Number.EPSILON
- add Number.MAX_SAFE_INTEGER
- add Number.MIN_SAFE_INTEGER

### CODE REVIEW

File: polyfills/Number/detect.js [hunk 1]
```diff
@@ -0,0 +1 @@
+Number.MAX_SAFE_INTEGER
```
Comment (jonathantneal) [inline polyfills/Number/detect.js:1]: Number will always be available.

**Comment by jonathantneal** [comment]
Whatever happened with this request?

**Comment by triblondon** [comment]
Sorry Jonathan, it's just a bit beyond my understanding, and it seems Juan had an issue with it.  It does also need conflicts resolved.  But I've got not problem with it in principle.

**Comment by triblondon** [comment]
@juandopazo @jonathantneal I'm going to close this request shortly, let me know if you'd like to keep it open.

**Pull request closed without merging**